File: 设计模式/设计模式/命令模式/单品/复合命令：执行多个命令/CompoundCommand.m
//
//  CompoundCommand.m
//  设计模式
//
//  Created by zachary spark on 2019/5/14.
//  Copyright © 2019 3g. All rights reserved.
//

#import "CompoundCommand.h"

@interface CompoundCommand()
@property(nonatomic,strong)NSMutableArray *Commands;

@end

@implementation CompoundCommand
- (instancetype)init:(NSMutableArray *)Commands{
    self = [super init];
    if(self){
        self.Commands = Commands;
    }
    return self;
}

//在我的复合命令中，调用执行多个命令就叫做复合命令
-(void)execute{
    
    for (id<DanPinCommandProtocol> command in self.Commands) {
        [command execute];
    }
    
}


@end

File: 设计模式-Swift/设计模式-Swift/命令模式/复合命令/CompoundCommand.swift
//
//  CompoundCommand.swift
//  设计模式-Swift
//
//  Created by zachary spark on 2019/5/14.
//  Copyright © 2019 3g. All rights reserved.
//

import UIKit

class CompoundCommand: DanPinCommandProtocol {
    
    private var commands:[DanPinCommandProtocol] = [DanPinCommandProtocol]()
    
    //执行多个命令
    var data:AnyObject?
    
    //定义个构造方法
    init(_ commands:[DanPinCommandProtocol]) {
      
        self.commands = commands
        
    }
    
    func execute() {
        
        for command in self.commands {
            command.execute()
        }
        
    }
    
    
}
